From 5ca9e9149514027822dbae392f2abd44b289d81b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:58 -0400
Subject: [usb] wusbcore: set the RPIPE bOverTheAirInterval for isoc endpoints

Message-id: <1410887463-20674-107-git-send-email-dzickus@redhat.com>
Patchwork-id: 95188
O-Subject: [RHEL7 PATCH 106/591] usb: wusbcore: set the RPIPE bOverTheAirInterval for isoc endpoints
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 1a7ff0e3673807ff6f38e7ae7b79e46a986fc578
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Tue Oct 1 14:04:34 2013 -0500

    usb: wusbcore: set the RPIPE bOverTheAirInterval for isoc endpoints

    This patch sets the RPIPE bOverTheAirInterval field for RPIPES which
    refer to isochronous endpoints.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-rpipe.c b/drivers/usb/wusbcore/wa-rpipe.c
index e3b913b..a76e2d9 100644
--- a/drivers/usb/wusbcore/wa-rpipe.c
+++ b/drivers/usb/wusbcore/wa-rpipe.c
@@ -361,8 +361,10 @@ static int rpipe_aim(struct wa_rpipe *rpipe, struct wahc *wa,
    epcd->bMaxSequence, 32U), 2U);
  rpipe->descr.bMaxDataSequence = epcd_max_sequence - 1;
  rpipe->descr.bInterval = ep->desc.bInterval;
- /* FIXME: bOverTheAirInterval */
- rpipe->descr.bOverTheAirInterval = 0; /* 0 if not isoc */
+ if (usb_endpoint_xfer_isoc(&ep->desc))
+  rpipe->descr.bOverTheAirInterval = epcd->bOverTheAirInterval;
+ else
+  rpipe->descr.bOverTheAirInterval = 0; /* 0 if not isoc */
  /* FIXME: xmit power & preamble blah blah */
  rpipe->descr.bmAttribute = (ep->desc.bmAttributes &
      USB_ENDPOINT_XFERTYPE_MASK);
-- 
1.7.1